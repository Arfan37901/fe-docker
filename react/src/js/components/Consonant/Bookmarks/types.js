import {
    bool,
    func,
    number,
} from 'prop-types';

// eslint-disable-next-line import/prefer-default-export
export const BookmarksType = {
    showBookmarks: bool,
    savedCardsCount: number,
    onClick: func.isRequired,
};
