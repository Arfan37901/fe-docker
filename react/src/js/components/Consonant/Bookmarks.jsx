import React from 'react';
import PropTypes from 'prop-types';

const Bookmarks = (props) => {
    const {
        selectedIco,
        unselectedIco,
        selected,
        onClick,
        qty,
    } = props;
    return (
        <button
            type="button"
            onClick={onClick}
            className={selected ? 'bookmarks bookmarks_selected' : 'bookmarks'
            }>
            <span className="bookmarks--ico-wrapper">
                {(selectedIco && unselectedIco) ?
                    <img
                        src={selected ? unselectedIco : selectedIco}
                        width="16"
                        alt=""
                        loading="lazy" /> :
                    <span className="bookmarks--ico" />
                }
                <span className="bookmarks--title">My favorites</span>
            </span>
            <span className="bookmarks--item-badge">{qty}</span>
        </button>
    );
};

export default Bookmarks;

Bookmarks.propTypes = {
    selectedIco: PropTypes.string,
    unselectedIco: PropTypes.string,
    selected: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
    qty: PropTypes.number,
};

Bookmarks.defaultProps = {
    selectedIco: '',
    unselectedIco: '',
    selected: false,
    qty: 0,
};
