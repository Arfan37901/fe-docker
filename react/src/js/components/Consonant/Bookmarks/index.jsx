import React, { useMemo } from 'react';
import classNames from 'classnames';

import { selector } from './selectors';
import { BookmarksType } from './types';
import { useConfigSelector } from '../Helpers/hooks';

const defaultProps = {
    showBookmarks: false,
    savedCardsCount: 0,
};

const Bookmarks = ({ showBookmarks, onClick, savedCardsCount }) => {
    const {
        label,
        selectIcon,
        unselectIcon,
    } = useConfigSelector(selector);

    const icon = showBookmarks
        ? selectIcon
        : unselectIcon;

    const bookmarkClass = classNames({
        bookmarks: true,
        bookmarks_selected: showBookmarks,
    });

    const iconStyles = useMemo(() => ({ backgroundImage: `url(${icon})` }), [icon]);

    return (
        <button
            tabIndex="0"
            type="button"
            onClick={onClick}
            data-testid="bookmarks"
            className={bookmarkClass}>
            <span className="bookmarks__icon-wrapper">
                <span
                    style={iconStyles}
                    className="bookmarks--icon"
                    data-testid="bookmarks--icon" />
                <span className="bookmarks__label">{label}</span>
            </span>

            <span
                data-testid="bookmarks__badge"
                className="bookmarks--item-badge">
                {savedCardsCount}
            </span>
        </button>
    );
};

Bookmarks.propTypes = BookmarksType;
Bookmarks.defaultProps = defaultProps;

export default Bookmarks;
