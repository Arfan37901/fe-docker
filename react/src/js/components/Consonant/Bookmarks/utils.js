// eslint-disable-next-line import/prefer-default-export
export const selector = ({
    bookmarks: {
        leftFilterPanel: { selectBookmarksIcon, unselectBookmarksIcon },
        i18n: {
            leftFilterPanel: { filterTitle },
        },
    },
}) => ({
    label: filterTitle,
    selectIcon: selectBookmarksIcon,
    unselectIcon: unselectBookmarksIcon,
});
