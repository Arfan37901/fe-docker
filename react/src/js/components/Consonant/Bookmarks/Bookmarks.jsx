import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { useConfig } from '../../../utils/hooks';

const Bookmarks = (props) => {
    const {
        selected,
        onClick,
        qty,
    } = props;

    const getConfig = useConfig();
    const title = getConfig('bookmarks', 'i18n.leftFilterPanel.filterTitle');
    const selectedIco = getConfig('bookmarks', 'leftFilterPanel.selectBookmarksIcon');
    const unselectedIco = getConfig('bookmarks', 'leftFilterPanel.unselectBookmarksIcon');

    const src = selected ? selectedIco : unselectedIco;

    const iconStyles = useMemo(() => ({
        backgroundImage: src ? `url(${src})` : '',
    }));

    return (
        <button
            data-testid="bookmarks"
            type="button"
            onClick={onClick}
            className={selected ? 'bookmarks bookmarks_selected' : 'bookmarks'}
            tabIndex="0">
            <span className="bookmarks--ico-wrapper">
                <span
                    style={iconStyles}
                    className="bookmarks--ico"
                    data-testid="bookmarks--ico" />
                <span className="bookmarks--title">{title}</span>
            </span>
            <span data-testid="bookmarks--item-badge" className="bookmarks--item-badge">{qty}</span>
        </button>
    );
};

export default Bookmarks;

Bookmarks.propTypes = {
    selected: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
    qty: PropTypes.number,
};

Bookmarks.defaultProps = {
    selected: false,
    qty: 0,
};
