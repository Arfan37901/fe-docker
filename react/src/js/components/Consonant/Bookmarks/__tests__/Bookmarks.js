import { screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Bookmarks from '../Bookmarks';

import {
    COUNT_LIST,
    DEFAULT_PROPS,
    WITHOUT_ICONS,
} from '../../Helpers/Testing/Constants/Bookmarks';

import makeSetup from '../../Helpers/Testing/Utils/Settings';

const setup = makeSetup(Bookmarks, DEFAULT_PROPS);

describe('Consonant/Bookmarks', () => {
    test('should renders correctly different bookmarks count', () => {
        COUNT_LIST.forEach((count) => {
            const { wrapper } = setup({ qty: count });

            const badgeElement = screen.getByTestId('bookmarks--item-badge');

            expect(badgeElement).toHaveTextContent(count);
            wrapper.unmount();
        });
    });

    test('should render unselectedIco', async () => {
        const {
            config: {
                bookmarks: {
                    leftFilterPanel: {
                        unselectBookmarksIcon,
                    },
                },
            },
        } = setup();

        const iconElement = screen.getByTestId('bookmarks--ico');

        expect(iconElement).toHaveStyle(`background-image: url(${unselectBookmarksIcon})`);
    });
    test('should render selectedIco', () => {
        const {
            config: {
                bookmarks: {
                    leftFilterPanel: {
                        selectBookmarksIcon,
                    },
                },
            },
        } = setup({ selected: true });

        const iconElement = screen.getByTestId('bookmarks--ico');

        expect(iconElement).toHaveStyle(`background-image: url(${selectBookmarksIcon})`);
    });
    test('shouldn`t have style object when selected === true && selectedIco didnt exists', async () => {
        setup({ selected: true }, { bookmarks: WITHOUT_ICONS });

        const iconElement = await waitFor(() => screen.getByTestId('bookmarks--ico'));

        expect(iconElement).not.toHaveStyle({ backgroundImage: "url('')" });
    });
    test('shouldn`t have style object when selected === false && unselectedIco didnt exists', async () => {
        setup({}, { bookmarks: WITHOUT_ICONS });

        const iconElement = await waitFor(() => screen.getByTestId('bookmarks--ico'));

        expect(iconElement).not.toHaveStyle({ backgroundImage: "url('')" });
    });

    describe('Check snapshots', () => {
        test('should renders correctly', () => {
            const { tree } = setup();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('Interaction with UI', () => {
        test('should call onChange', () => {
            const { props: { onClick } } = setup();

            const boorkmarksElement = screen.getByTestId('bookmarks');

            fireEvent.click(boorkmarksElement);

            expect(onClick).toBeCalled();
        });
    });
});
