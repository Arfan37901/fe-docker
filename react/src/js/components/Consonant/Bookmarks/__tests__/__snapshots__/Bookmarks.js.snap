// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Bookmarks Check snapshots should renders correctly 1`] = `
<button
  className="bookmarks"
  data-testid="bookmarks"
  onClick={[MockFunction]}
  tabIndex="0"
  type="button"
>
  <span
    className="bookmarks--ico-wrapper"
  >
    <span
      className="bookmarks--ico"
      data-testid="bookmarks--ico"
      style={
        Object {
          "backgroundImage": "url(http://placehold.it/150x150/red)",
        }
      }
    />
    <span
      className="bookmarks--title"
    >
      i18n.leftFilterPanel.filterTitle
    </span>
  </span>
  <span
    className="bookmarks--item-badge"
    data-testid="bookmarks--item-badge"
  >
    0
  </span>
</button>
`;
